src/components/ERACommons.jsx

```diff
@@ -0,0 +1,187 @@
+    const expProp = find({'propertyKey':'eraExpiration'})(properties);
+    const isAuthorized = isNil(authProp) ? false : getOr(false,'propertyValue')(authProp);
+    const expirationCount = isNil(expProp) ? 0 : expirationCountFromDate(getOr(0,'propertyValue')(expProp));
+    const nihValid = isAuthorized && expirationCount > 0;
+    props.onNihStatusUpdate(nihValid);
+    setAuthorized(isAuthorized);
+    setExpirationCount(expirationCount);
+    setEraCommonsId(isNil(eraCommonsId) ? '' : eraCommonsId);
+  };
+
+  /**
+   * This useEffect is intended to run once after a user has been redirected back to the current page from a
+   * successful NIH authentication.
+   */
+  useEffect(() => {
+    const init = async () => {
+      // If we have a token to verify, save it before getting user info
+      if (props.location !== undefined && props.location.search !== '') {
+        await saveNIHAuthentication(props.location.search);
+      }
+    };
+    init();
+    // Note that we do not want props.location as a dependency here since it does not change after render.
+  }, []);
+
+  /**
+   * This useEffect is intended to populate state from either the provided researcher object or the current user.
+   */
+  useEffect(() => {
+    const fetchData = async () => {
+      // In the read-only case, we are provided a researcher object and do not need to query for the current user
+      if (props.readOnly && props.researcherProfile) {
+        setResearcherProfile(props.researcherProfile);
+        const propsData = researcherProfile.properties;
+        setCommonResearcherPropertyState(propsData, researcherProfile.eraCommonsId);
+      } else {
+        await getUserInfo();
+      }
+    };
+    fetchData();
+  }, []);
+
+  const onMouseEnter = () => {
+    setHovered(true);
+  };
+
+  const onMouseLeave = () => {
+    setHovered(false);
+  };
+
+  const saveNIHAuthentication = async (searchArg) => {
+    const rawToken = qs.parse(searchArg);
+    const decodedToken = await decodeNihToken(rawToken);
+    if (isNil(decodedToken)) {
+      setNihError(true);
+      return;
+    }
+    // We need to rewrite the payload to match the expected format in Consent.
+    const nihPayload = {
+      'linkedNihUsername': `${decodedToken.eraCommonsUsername}`,
+      'linkExpireTime': `${decodedToken.exp}`,
+      'status': true
+    };
+    const updatedUserProps = await AuthenticateNIH.saveNihUsr(nihPayload);
+    setResearcherProfile(updatedUserProps);
+    setCommonResearcherPropertyState(updatedUserProps, decodedToken.eraCommonsUsername);
+    document.getElementById('era-commons-id').scrollIntoView(
+      {block: 'start', inline: 'nearest', behavior: 'smooth'}
+    );
+  };
+
+  const redirectToNihLogin = async () => {
+    const destination = window.location.origin + '/' + props.destination + '?nih-username-token=<token>';
+    const nihUrl = `${ await Config.getNihUrl() }?${qs.stringify({ 'return-url': destination })}`;
+    window.location.href = nihUrl;
```

Comment:
```suggestion
    props.history.push(nihUrl);
```

Should we use the React mechanism for redirecting instead?

Reply:
This only redirects internally, it doesn't redirect externally ... i.e., it tacks on the `nihUrl` to the base url which ends up failing. I don't see another mechanism for doing this, but would love to see something other than the `window.location.href` approach